addons/purchase_requisition/views/product_views.xml

```diff
@@ -35,6 +35,12 @@
                     <field name="purchase_requisition" widget="radio"/>
                 </group>
             </group>
+            <xpath expr="//field[@name='seller_ids']/tree//field[@name='product_id']" position="after">
+                <field name="purchase_requisition_id" attrs="{'invisible': [('purchase_requisition_id', '=', False)]}" optional="hide"/>
+            </xpath>
+            <xpath expr="//field[@name='variant_seller_ids']/tree//field[@name='product_id']" position="after">
+                <field name="purchase_requisition_id" attrs="{'invisible': [('purchase_requisition_id', '=', False)]}" optional="hide"/>
+            </xpath>
```

Comment:
When you will apply my first comment, you will need only one xpath, so don't forget to modify this too.

Reply:
Also, I'm not sure this conditional invisible attribute makes any sense in a list view.
What you try to achieve here ?